src/settings/toml/manifest.rs

```diff
@@ -115,6 +120,35 @@ impl Manifest {
         Ok(template_config)
     }
 
+    pub fn worker_name(&self, env_arg: Option<&str>) -> String {
+        if let Some(environment) = self.get_environment(env_arg).unwrap_or_default() {
+            if let Some(name) = &environment.name {
+                return name.clone();
+            }
+            if let Some(env) = env_arg {
+                return format!("{}-{}", self.name, env);
+            }
+        }
+
+        self.name.clone()
+    }
+
+    fn route_config(&self) -> RouteConfig {
+        RouteConfig {
+            workers_dev: self.workers_dev,
+            route: self.route.clone(),
+            routes: self.routes.clone(),
+            zone_id: self.zone_id.clone(),
+        }
+    }
+
+    pub fn deploy_target(&self, env: Option<&str>) -> Result<DeployTarget, failure::Error> {
```

Comment:
this method will be used in place of `get_target` for the deploy step in publish. will likely also add something similar for `build_target` that i predict will be less convoluted.